wa/framework/configuration/tree.py

```diff
@@ -69,14 +69,20 @@ def name(self):
     def is_leaf(self):
         return not bool(self.children)
 
-    def __init__(self, config, parent=None):
+    def __init__(self, config, parent=None, group=None):
         super(SectionNode, self).__init__(config, parent=parent)
         self.workload_entries = []
         self.children = []
+        self.group = group
 
-    def add_section(self, section):
-        new_node = SectionNode(section, parent=self)
-        self.children.append(new_node)
+    def add_section(self, section, group=None):
+        # Each level is the same group, only need to check first
+        if not self.children or group == self.children[0].group:
+            new_node = SectionNode(section, parent=self, group=group)
+            self.children.append(new_node)
+        else:
+            for child in self.children:
+                new_node = child.add_section(section, group)
```

Comment:
What happens if there are more than two groups?

Reply:
The recursive method will continue until it finds a matching group to add to or will create a new a layer for the group.

Reply:
ah, of course; missed that.